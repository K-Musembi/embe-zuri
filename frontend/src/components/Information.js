import React from "react";
import { useEffect, useState } from "react";
import { API_URL } from "../config";

const Information = ({ prediction }) => {

    const [image, setImage] = useState(null);
    const [info, setInfo] = useState(null);

    // useEffect is used when a spcific action must take place once a page is rendered
    // useEffect dependencies should be included, if any e.g. prediction
    // to access data from the useEffect hook, use a useState hook e.g. info
    useEffect(() => {
        const fetchInfo = async () => {
            const response = await fetch(`${API_URL}/information`, {
                method: "POST",
                headers: { "Content-Type": "application/json" },
                body: JSON.stringify({ prediction })
            });
            if (response.ok) {
                const data = await response.json();
                setImage(`data:image/jpeg;base64,${data.image}`);
                setInfo(data);
            } else {
                console.error("information not available");
            }
        };
        fetchInfo();
    });

    return (
        <div className="information-panel">
            <div className="left-panel">
                <div className="description">
                    <h2>{info.name}</h2>
                    <p>{info.description}</p>
                </div>
                <div className="appearance-causes">
                    {info.name === "Healthy" ? (
                        <h3>How a healthy mango tree appears</h3>
                    ) : (
                        <h3>Main Causes</h3>
                    )}
                    <p>{info.causes}</p>
                </div>
            </div>
            <div className="right-panel">
                <div className="image">
                    {image ? <img src={image} alt="Loaded" /> : <p>Loading image...</p>}
                </div>
                <div className="tips-remedies">
                    {info.name === "Healthy" ? (
                        <h3>Maintaining a healthy farm</h3>
                    ) : (
                        <h3>Prevention Measures</h3>
                    )}
                    {info.remedies.map((remedy, index) => (
                        <div className="list" key={index}>{remedy}</div>
                    ))}
                </div>
            </div>
            <h1>{prediction}</h1>
        </div>
    );
};

export default Information;
