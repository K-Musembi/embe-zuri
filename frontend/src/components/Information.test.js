import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Information from './Result';

// Mock fetch globally
global.fetch = jest.fn(() =>
  Promise.resolve({
    ok: true,
    json: () => Promise.resolve({ data: 'mocked data' }),
  })
);

describe('Information Component', () => {
  const mockResult = {
    name: "Healthy",
    causes: ["Main causes"],
    description: ["Main description"],
    remedies: ["Use antifungal spray", "Ensure proper ventilation"],
  };

  it('renders correctly with result', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Information result={mockResult} onPrediction={jest.fn()} />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
