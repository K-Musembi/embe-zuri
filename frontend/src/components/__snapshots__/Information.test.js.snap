// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information Component renders correctly with result 1`] = `
<div
  className="result-panel"
>
  <div
    className="result-container"
  >
    <div
      className="prediction-outcome"
    >
      <h2>
        Prediction outcome
      </h2>
      <h3
        className="not-healthy"
      >
        Not healthy: Disease detected
      </h3>
      <h3>
        Probable disease: 
      </h3>
      <h3>
        Confidence: 
        %
      </h3>
    </div>
    <div
      className="recommendations"
    >
      <h2>
        Recommendations
      </h2>
      <div
        className="remedy"
      >
        Use antifungal spray
      </div>
      <div
        className="remedy"
      >
        Ensure proper ventilation
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    More Information
  </button>
</div>
`;

exports[`Information Component renders no-information state correctly 1`] = `
<div
  className="information-panel"
>
  <div
    className="no-information"
  >
    <p>
      No information available. Please upload image again.
    </p>
  </div>
</div>
`;

exports[`Result Component renders correctly with result 1`] = `
<div
  className="result-panel"
>
  <div
    className="result-container"
  >
    <div
      className="prediction-outcome"
    >
      <h2>
        Prediction outcome
      </h2>
      <h3
        className="not-healthy"
      >
        Not healthy: Disease detected
      </h3>
      <h3>
        Probable disease: 
      </h3>
      <h3>
        Confidence: 
        %
      </h3>
    </div>
    <div
      className="recommendations"
    >
      <h2>
        Recommendations
      </h2>
      <div
        className="remedy"
      >
        Use antifungal spray
      </div>
      <div
        className="remedy"
      >
        Ensure proper ventilation
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    More Information
  </button>
</div>
`;
